### PR TITLE
yamcs-xtce: Add support for IndirectParameterRefEntry

### PR DESCRIPTION
**Pull request by jdiez17**
Needed for @xpromache's suggested approach of using IndirectParamRefEntry to implement "deduced" parameter value types as defined in PUS.

I'll sign the CLA shortly.

### CODE REVIEW

**Comment by xpromache** [comment]
Thanks Jose for the pull request. Can you please sign the CLA
https://yamcs.org/static/Yamcs_Contributor_Agreement_v2.0.pdf
and send it to yamcs@spaceapplications.com

thx

**Comment by jdiez17** [comment]
I submitted the CLA.

File: yamcs-xtce/src/main/java/org/yamcs/xtce/xml/XtceStaxReader.java [hunk 1]
```diff
@@ -2552,6 +2552,31 @@ private void readEntryList(SpaceSystem spaceSystem, Container container, MetaCom
 
     }
 
+    private SequenceEntry readIndirectParameterRefEntry(SpaceSystem spaceSystem) throws XMLStreamException {
+        log.trace(ELEM_INDIRECT_PARAMETER_REF_ENTRY);
+        checkStartElementPreconditions();
+
+        String aliasNameSpace = readMandatoryAttribute(ATTR_ALIAS_NAME_SPACE, xmlEvent.asStartElement());
```
[the remark this answers is not in the document]
Comment (jdiez17) [reply yamcs-xtce/src/main/java/org/yamcs/xtce/xml/XtceStaxReader.java]: Done. I guess I only looked at the XML and missed the description.

File: yamcs-xtce/src/main/java/org/yamcs/xtce/xml/XtceStaxReader.java [hunk 2]
```diff
@@ -2552,6 +2552,31 @@ private void readEntryList(SpaceSystem spaceSystem, Container container, MetaCom
...
+        while (true) {
+            xmlEvent = xmlEventReader.nextEvent();
+
+            if (isStartElementWithName(ELEM_LOCATION_IN_CONTAINER_IN_BITS)) {
+                readLocationInContainerInBits(indirectParameterRefEntry);
+            } else if (isStartElementWithName(ELEM_PARAMETER_INSTANCE)) {
+                ParameterInstanceRef ref = readParameterInstanceRef(spaceSystem, null);
+                indirectParameterRefEntry.setParameterRef(ref);
+            } else if (isEndElementWithName(ELEM_INDIRECT_PARAMETER_REF_ENTRY)) {
+                return indirectParameterRefEntry;
```
[the remark this answers is not in the document]
Comment (jdiez17) [reply yamcs-xtce/src/main/java/org/yamcs/xtce/xml/XtceStaxReader.java:2576]: Done.

File: yamcs-xtce/src/test/java/org/yamcs/xtce/IndirectParameterRefTest.java
[the remark this answers is not in the document]
Comment (jdiez17) [reply yamcs-xtce/src/test/java/org/yamcs/xtce/IndirectParameterRefTest.java:1]: Done. You're welcome! :)

**Comment by jdiez17** [comment]
I implemented the requested changes and ran `mvn test` on `yamcs-xtce` and `yamcs-core`.

I got this test error, but I think it's unrelated to these changes:

```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   RdbEngineTest.testOpenClose:61 » DatabaseCorruption No (version, maxTbsIndex) record found in the metadata
[ERROR]   RdbEngineTest.testOpenCloseWithTableDrop:76 » Yarch Got exception when reading the table definition from /tmp/yamcs-data/yarchtest_StreamMergeTest.rdb: 
[INFO] 
[ERROR] Tests run: 899, Failures: 0, Errors: 2, Skipped: 15
```

**Comment by xpromache** [comment]
Perfect, thank  you very much. 

I finally have a good idea on how to deal with the ParameterInstanceRef: I will introduce a qualifier to define exactly what that instance field means (positive, negative counting forward backward, etc).
Then when creating it from the Xtce reader we have the option to set it differently for the IndirectParamRefEntry and for the older ones to maintain compatibility.
We have also the option to change the behavior using a custom AncillaryDataSet annotation or just adding some options to the XTCE loader  (later on when someone will request this).  

I have to find a good name for that qualifier (it will be an enum field in the ParameterInstanceRef class) and then I will merge your pull request.

**Pull request closed without merging**